ktor-client/ktor-client-features/ktor-client-json/ktor-client-serialization/common/test/ContextualSerializationTest.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+import io.ktor.client.call.*
+import io.ktor.client.features.json.serializer.*
+import io.ktor.utils.io.core.*
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+import kotlinx.serialization.modules.*
+import kotlin.reflect.*
+import kotlin.reflect.typeOf
+import kotlin.test.*
+
+
+@Serializable
+data class UserData(val id: Int, val name: String)
+
+object UserDataSerializer : KSerializer<UserData> {
```

Comment:
I am not really aware of Ktor code style, so can't say whether this is an intentional approach or not.

But both in serialization and coroutines we define such classes within test classes to reduce autocompletion and search by symbol pollution

Reply:
I'm not sure if we need to define it inside, but we can mark it with private at least.

Reply:
Again, this is opinionated, feel free to ignore my comment if you find your approach more convenient.

Declaring it inside will hide `UserDataSerializer` and `UserData` from autocomplete in other tests